.buildkite/commands/run-ui-tests.sh

```diff
@@ -47,5 +47,10 @@ if [[ $TESTS_EXIT_STATUS -eq 0 ]]; then
 else
   echo "The UI Tests, ran during the '🔬 Testing' step above, have failed."
   echo "For more details about the failed tests, check the logs under the '🔬 Testing' section and the \`.xcresult\` and test reports in Buildkite artifacts."
+
+  JUNIT_REPORT="build/results/report.junit"
+  if [ -f "$JUNIT_REPORT" ]; then
+    xsltproc --stringparam step_title "$BUILDKITE_LABEL" ".buildkite/commands/junit-failures-to-buildkite-annotation.xslt" "$JUNIT_REPORT" | buildkite-agent annotate --style error --context "ui-tests-$TEST_NAME-$DEVICE"
+  fi
```

Comment:
@AliSoftware, you [commented](https://github.com/wordpress-mobile/WordPress-iOS/pull/18952/#issuecomment-1169332988):

> the difference being in the real case, we do a command to cd build/results/ && zip … to zip the xcresult bundle… but don't revert the PWD back after that.

I can see the `&& cd -` in the unit tests command script via [`25e8b00`](https://github.com/wordpress-mobile/WordPress-iOS/pull/18952/commits/25e8b0063a0057bf3a1c596d0947444fcab3e4b3), but I can't find one here, in the UI tests command ([see line 42](https://github.com/wordpress-mobile/WordPress-iOS/pull/18952/files#diff-bd954ba2ef49cba9cd9950b5197be8e22f9c95f5482f03d0186eaf9404cda0e8R42)).

Reply:
Dammit, good point, forgot to add it here as well, nice catch!

@mokagio Can I let you add it, then tweak https://github.com/wordpress-mobile/WordPress-iOS/pull/18955 in your day to trigger a build that would test the case of when there's an actual UI Test failure (basically, probably only keep 63e73e8037b91f883657547743bd707ea1ea1f05, but revert bc15c6c52ab8ffa8f9f9a0c3f081efa0448601ad, to still run the full pipeline this time, but with intentional test failures)? 🥺 🙏 

That should allow us to validate that this `&& cd -` was really what was missing and that it solves the previous `exit 1` error when test failure occurs, in both Unit and UI tests

Reply:
Done, the build is running here: https://buildkite.com/automattic/wordpress-ios/builds/8636#0181ad9f-9761-408c-823e-72812b8951ed

Leaving this WIP comment in case I don't get a chance to see the build finish and wrap this up.